chain/chain/src/chain.rs

```diff
@@ -282,6 +282,7 @@ impl Chain {
                         &genesis.header(),
                         // genesis height is considered final
                         chain_genesis.height,
+                        true,
```

Comment:
I propose we either have an enum with readable variants or separate `new` methods, so the intention is clear. Passing `true` or `false` as a parameter requires the reader to dig into the implementation/signature matching the order of parameters to learn what exactly it is responsible for.